control/utils.py

```diff
@@ -175,6 +176,43 @@ def is_valid_nqn(nqn):
 
         return (0, os.strerror(0))
 
+    def get_directory_with_fsid(logger, config, ceph_utils, conf_section, conf_field, def_value, suffix, should_create):
```

Comment:
If my understanding is correct, Cephadm mounts external directories inside the container to meet the requirements of the service. So that's how the responsibility chain goes.

Services (like nvmeof gw) shouldn't try to circumvent that, but just expect that Cephadm provide assets at required locations. That means that services, for example, don't have to deal with the fsid, since that's something that Cephadm does itself.

Additionally, Cephadm now has a specific set of CLI/API commands to deal with secrets, like TLS keys.

Reply:
I don't want to rely on Cephadm. And as we already had the information ready for previous features I just used it here.

Reply:
Why? That's what Cephadm does. Services shouldn't have to deal with stuff that the orchestrator does for them. Trying to outsmart the inner layers of system usually lead, in the best case, to unnecessary complexity, and in the worst, to issues. Anyone has their own scope of action, and services shouldn't try to guess directory locations, but expect that things are properly set by the orchestrator.

Reply:
As I said, I didn't want to rely on cephadm as I don't have it in development environemnt and we already get information from Ceph, including FSID, in other places so I haven't introduce anything new here. I just assembled already exiting pieces of information.

Reply:
In any case, the cephadm can override this entire function. All it's need to do is to set the path in the configuration to the required directory. If this is done the gateway will just take the path as is and wouldn't do any processing with it.